In [11]:
import numpy as np

# Seminar 2 #

 1. **Fundamental of Hydrostatics**
 2. **Forces on submerged surfaces**
 3. **Pressure and buoyancy**




### Problem 1
Rectangular container of a width of 1 m and a length of 2 m is filled to a depth of 0.6 m with oil.
Calculate the pressure at the bottom of the tank. What is the weight of the oil? Assume specific weight of the oil is 8000 N/m$^3$.



### Solution of Problem 1

The liquid is OIL in the problem. The main equations are:

$$ P = \rho \cdot g \cdot h$$

$$
P_{abs} = P_{atm} + P_{gage}
$$

$$ \gamma = \rho \cdot g = \frac{M}{V}\cdot g = \frac{W}{V\cdot g} \cdot g = \frac{W}{V} $$

$$
W = \gamma  V
$$



with $\gamma$ = specific weight (N/m<sup>3</sup>), and $M$ is mass (Kg), $W$ is weight (N) and $V$ volume (m<sup>3</sup>). 

other available information are:

In [3]:
#Given,

ga1 = 8500 # N/m^3, specific weight of oil
W1 = 1 # m, width of the tank
L1 = 2 # m, length of the tank
H1 = 0.6 # m, oil depth in the tank
g = 9.81 # m/s^2, earth's gravity
P1_atm = 101000 # N/m^2, Standard atmospheric pressure

# interim calculation
V1 = L1*W1*H1 # m^3, filled volume of the tank
P1_oi1 = ga1*H1 #N/m^2, gamma*h

#Calculations
P1_abs = P1_atm + P1_oi1
W1_o= ga1* V1

# output
print("The pressure of the tank is {0:1.2f}".format(P1_abs),"N/m\u00b2", "\n")
print("The Weight of the oil in the tank is {0:1.2f}".format(W1_o),"N")  

The pressure of the tank is 106100.00 N/m² 

The Weight of the oil in the tank is 10200.00 N



### Problem 2 - Vertically submerged tank

Given the following rectangular gate with $h_1 = 1$ m, $L = 2$ m and $D=3$ m  :

<img src="figures/S2-fig1.png" alt="Rectangular Tank" width="420">

Determine for water:

1. Pressure at the bottom of the gate
2. resultant hydrostatic forces 
3. depth at which the resultant force acts

### Solution of Problem 2 ###

The relevant equations are (check slides 18 - 20 of L2):

$$
F_r = \rho\cdot g \cdot h_g \cdot A
$$

$$
h_g = h_1 + \frac{D}{2}
$$
and

$$
h_p = \Bigg(\frac{I_g}{A\cdot h_g}\Bigg) + h_g
$$

with $I_g = \frac{L \cdot D^3}{12}$.

Information provided in the problem are:



In [2]:
# Given are

dy2 = 1000 # kg/m^3, water density
g2 = 9.81 # m/s^2
h2_1 = 1 # m, height from surface to gate top
D2 = 3 # m, depth of the tank
L2 = 2 # m, Length of tank

# interim calculation
A2 = L2*D2 # m^2, Area of tank
h2_g = h2_1 + D2/2 # m, height from top to centroid
I2_g = L2*D2**3/12 # m^4, second moment of area


# calculation
P2_bot = dy2 * g2* (h2_1 + D2) # N/m^2, P = rho.g.h, h = (h1+D) see fig.
F2_r = dy2 *g2*h2_g*A2 # N, Resultant force
h2_p = I2_g/(A2*h2_g)+ h2_g


# output
print("The pressure of tank bottom is {0:1.4f}".format(P2_bot),"N/m\u00b2", "\n")
print("The resultant force in the tank is {0:1.2f}".format(F2_r),"N",  "\n")  
print("The location of resultant force is {0:1.2f}".format(h2_p), "m")

P2_bot

The pressure of tank bottom is 39240.0000 N/m² 

The resultant force in the tank is 147150.00 N 

The location of resultant force is 2.80 m


39240.0

### Problem 3 - Vertically submerged tank

A lock on a canal is sealed by a gate that is 3.0 m wide (see fig below). The gate is perpendicular to the sides
of the lock. When the lock is used there is water on one side of the gate to a depth of 3.5 m,
and 2.0 m on the other side. 

(a) What is the hydrostatic force of the two sides of the gate? 

(b) At what height from the bed do the two forces act?

(c) What is the magnitude of the overall
resultant hydrostatic force on the gate and at what height does it act?

<img src="figures/S2-fig2A.png" alt="Rectangular Tank" width="420"> 

### Solution of problem 3 ###

The relevant equations are (check slides 18 - 20 of L2).

$$
F_r = \rho\cdot g \cdot h_g \cdot A
$$


Then, we draw the pressure diagram

<img src="figures/S2-fig2B.png" alt="Rectangular Tank" width="350">


Since both pressure diagram are triangular, $F_r$ acts at 1/3 from the bases - this is only true when pressure diagram is simple such as triangular in this case. 

Other information are:



In [4]:
# Given

W3 = 3 # m, Gate width
h_up = 3.5 # m, upstream water height
h_dn = 2 # m, downstream water height
dy3 = 1000 # kg/m^3, density of water
g3 = 9.81 # m/s^2, gravity

# interim calculation 
hg_up = h_up/2 # m, centroid of surface upstream
hg_dn = h_dn/2 # m, centroid of surface downstream
A_up = W3*h_up # m^2, area upstream
A_dn = W3*h_dn # m^2 area downstream

# calculation (a) and (b)
F_up = dy3*g3*hg_up*A_up  # N, resultant force up stream
F_dn = dy3*g3*hg_dn * A_dn # N, resultant force up stream
y_up = 1/3*h_up # m, Upstream location of centre of pressure from bottom.
y_dn = 1/3*h_dn # m, downstream Location of centre of pressure from base.

# output
print("The resultant force upstream  is: {0:1.2E}".format(F_up),"N", "\n")
print("The resultant force downstream  is: {0:1.2E}".format(F_dn),"N", "\n")
print("The location of resultant force upstream from the bottom  is {0:1.2f}".format(y_up),"m",  "\n")  
print("The location of resultant force downstream from the bottom  is {0:1.2f}".format(y_dn),"m", )

The resultant force upstream  is: 1.80E+05 N 

The resultant force downstream  is: 5.89E+04 N 

The location of resultant force upstream from the bottom  is 1.17 m 

The location of resultant force downstream from the bottom  is 0.67 m


In [5]:
# Solution 3C

Fr_o = F_up - F_dn # N, +ve means the resultant force is upstream

# Moment F * y about O, at the base (see fig)

M_up = F_up * y_up # N-m, moment in upstream
M_dn = F_dn * y_dn # N-m, moment in downstream

# Location of Resultant force (Moment balance equation)
y_r = (M_up - M_dn)/Fr_o # m, moment in the system is conserved

#output
print("The overall resultant force  is: {0:1.2E}".format(Fr_o),"N", "\n")
print("The location of overall resultant force  is: {0:1.2f}".format(y_r),"m")

The overall resultant force  is: 1.21E+05 N 

The location of overall resultant force  is: 1.41 m


### Problem 4 - inclined submerged surface ###

A sewer discharges to a river. At the end of the sewer is a circular gate with a diameter ($D$) of 0.6 m. The gate is inclined at an angle of 45 $^\circ$ to the water surface. The top edge of the gate is 1.0 m below the surface. Calculate
			
(a) the resultant force on the gate caused by the water in the river


(b) the vertical depth from the water surface to the centre of pressure.

<img src="figures/S2-fig3.png" alt="problem 4" width=300>



### Solution of Problem 4 ###

Reference lecture slides 25-27. Important equations are:

$$
h_g = h_1 + \frac{h_{gate}}{2}
$$

$$
F_r = \rho\cdot g \cdot h_g \cdot A
$$

$$
L_p = \frac{I_g}{A\cdot L_g} + L_g
$$
with $I_g = \pi R^4/4$ and $L_g = h_g/\sin\alpha $. $L_p$ and $h_p$ can be similarly obtained.

<img src="figures/S2-fig3a.png" alt="problem 4" width=300>


In [9]:
# Given
h4_1 = 0.5 # m, free surface height
d4 = 0.6 # m, diameter circular gate
apa = 30 # degrees, inclined angle
dy4 = 1000 # kg/m^3, density water
g4 = 9.81 # m/s^2, gravity

# interim calculation
h_gate = d4*np.sin(apa*np.pi/180) # m, check np.sin and np. pi and why * pi/180
A4 = np.pi/4*(d4)**2 # m^2 area of gate
h4_g = h4_1 + h_gate/2 # m, location of surface centroid

# calculation (a)
F4_r = dy4*g4*h4_g*A4

#output
print("The resultant force  is: {0:1.2f}".format(F4_r),"N") 

The resultant force  is: 1802.91 N


In [10]:
# solution 4(b) location h_p

# interim calculation
L4_g = h4_g/np.sin(45*np.pi/180) # m, inclined length from centroid
R4 = d4/2 # m , radius of the gate
I4_g = np.pi*R4**4/4 # m^4 second moment of area
L4_p = I4_g/(A4*L4_g) + L4_g # incline length from centre of force

#calculation
h4_p = L4_p*np.sin(45*np.pi/180) # vertical height to centre of force

#Output
print("The vertical depth to the centre of force  is: {0:1.2f}".format(h4_p),"m") 



The vertical depth to the centre of force  is: 0.67 m


### Problem 5 - curved submerged surface ###

An open tank which is 4.0 m wide at the top contains
oil to a depth of 3.4 m as shown in figure. The
bottom part of the tank has curved sides which have
to be bolted on. To enable the force on the bolts to be determined, calculate the magnitude of the resultant hydrostatic force (per metre length) on the curved surfaces and its angle to the horizontal. The curved sections are a quarter of a circle of 1.5 m radius, and the oil density is 800 kg/m<sup>3</sup>.

<img src="figures/S2-fig4.png" alt="problem 5" width=300> 



### Solution of Problem 5 ###

Reference lecture slides 30-31. Important equations are:

$$
h_g = (D - R) + \frac{R}{2}
$$

<img src="figures/S2-A4a.png" alt="problem 4" width=300>

$$
F_v = \rho\cdot g \cdot h_g \cdot A
$$

$$
F_h = \rho\cdot g \cdot V
$$

$$
F_r = \sqrt{F_h^2 + F_v^2}
$$

and

$$
\tan\phi = \frac{F_v}{F_h}
$$



In [11]:
# Given
dy5_o = 800 # kg/m^3, density of oil
W5 = 4 # m, tank width
D5 = 3.4 # m, Depth of wetted surface
L5 = 1 # m, length of surface see fig left in question
R5_p = 1.5 # m, Curved section radius 
g5 = 9.81 # m^2/s, gravity 

# interim calculation
A5_p = R5_p*L5 # m^2, projected curved area
h5_g = (D5 - R5_p) + R5_p/2
V5 =  np.pi/4*R5_p**2*L5 + R5_p*L5*(D5-R5_p) # m^3, circular volume + rectangular volume
# Circular vol = pi/4 * R^2 * L and Rect. vol = R*L*(D-R)

# Calculations 
F5_v = dy5_o*g5*h5_g*A5_p # N, Force vertical
F5_h = dy5_o*g5*V5 # N, Force horizontal
F5_r = np.sqrt(F5_h**2+F5_v**2) # N, Resultant force
phi_r = np.tanh(F5_v/F5_h) # rad, angle with horizontal surcface 

#output
print("The horizontal force  is: {0:1.2f}".format(F5_h),"N", "\n")
print("The vertical force  is: {0:1.2f}".format(F5_v),"N", "\n")
print("The resultant force  is: {0:1.2f}".format(F5_r),"N", "\n")
print("The angle of resultant force to the horizontal : {0:1.2f}".format(phi_r),"rad", "\n")
print("The angle of resultant force to the horizontal : {0:1.2f}".format(phi_r*180/np.pi),"deg", "\n")

The horizontal force  is: 36235.36 N 

The vertical force  is: 31195.80 N 

The resultant force  is: 47814.01 N 

The angle of resultant force to the horizontal : 0.70 rad 

The angle of resultant force to the horizontal : 39.92 deg 



### Problem 6 - Forces and Buoyancy

A pipe carrying natural gas is to be laid in seabed. The
weight of the pipe is 2360 N per metre length and its outside diameter is 1.0 m. The weight of the gas can be ignored. The density of sea water is 1025kg/m<sup>3</sup>. Determine 

a) whether the pipe will
remain on the sea bed or float. 

b) If it does float,
what force would be required to hold the pipe on
the sea bed?

<img src="figures/S2-fig6.png" alt="problem 6" width=200> 

### Solution problem 6 ###

The relevant equations are in slides L02 37-38. They are:

$$
F_v = \rho\cdot g\cdot V
$$

with volume $V$ 

**Floating** if
$$
F_v \geq W
$$
and **sinking** if

$$
F_v < W
$$

Net Force 
$$
F_n = F_v - W
$$



In [9]:
# Given 

W6 = 2360 # N, weight of pipeline
d6 = 1 # m, diameter of pipe
dy6_s = 1025 # kg/m^3, seawater density 
g6 = 9.81 # m/s^2, Gravity force

#interim calculation
V6 = np.pi/4*d6**2 # m^3/m = vol./length, 
F6_v = dy6_s*g6*V6 # N/m, buoyancy force/length

# calculation
if F6_v >= W6:
    print("It is floating")
else:
    print("It is Sinking") 

F6_net = F6_v - W6

# output
print("The force required to hold the pipe in sea-bed is:{0:1.2e}".format(F6_net), "N/m")
print("The force required to hold the pipe in sea-bed is:{0:1.2e}".format(F6_v), "N/m")

It is floating
The force required to hold the pipe in sea-bed is:5.54e+03 N/m
The force required to hold the pipe in sea-bed is:7.90e+03 N/m


### Assignment Problem 1 - Basic pressure calculation

Rectangular container of a width of 1.5 m and a length of 2.0 m is filled to a depth of 1.0 m with oil. If the mass of oil is 2000 Kg, Calculate:

a) the pressure at the bottom of the tank. 

b) the weight of the oil?



### Assignment Problem 2 - Vertically submerged body- pressure calculation

A rectangular culvert (a large pipe) 1.8 m wide by 1.0 m high discharges to a river. At the end
of the culvert is a rectangular gate which seals off the culvert when the river is in flood (as in
Fig.). The gate hangs vertically from hinges at the top. If the flood level in the river rises to
1.9 m above the top of the gate, calculate the magnitude and location of the resultant hydrostatic
force on the gate caused by the water in the river.

<img src="figures/S2-A2.png" alt="Assignment 2" width ="400">

### Assignment Problem 3 - Inclined submerged body- pressure calculation

A sewer discharges to a river. At the end of the sewer is a circular gate with a diameter ($D$) of 0.6 m. The gate is inclined at an angle of 25 $^\circ$ to the water surface. The top edge of the gate is 1.0 m below the surface. Calculate
			
(a) the resultant force on the gate caused by the water in the
				river
(b) the vertical depth from the water surface to the centre of pressure.

(c) compare your result with that of problem 4, and provide your opinion.

<img src="figures/S2-fig3.png" alt="problem 4" width=400>


### Assignment Problem 4 - Curved submerged body- pressure calculation

A surface consists of a quarter of a circle of radius 2.0 m (see Fig.). It is located with its top edge
1.5 m below the water surface. Calculate the magnitude and direction of the resultant force on
the upper surface.

<img src="figures/S2-A4.png" alt="problem 4" width=400>

			
